src/sockets/sockets.ts

```diff
@@ -3099,6 +3117,17 @@ function roomChatFromClient(data) {
     senderSocket.emit('roomChatToClient', data2);
     return;
   }
+  if (!chatSpamFilter.chatRequest(data.username)) {
+    const senderSocket =
+      allSockets[getIndexFromUsername(allSockets, data.username, true)];
+    const data3 = {
+      message: 'Woah chill buddy!! Lets take a breather..yeah?',
+      classStr: 'all-chat-text-red',
+      dateCreated: new Date(),
+    };
+    senderSocket.emit('roomChatToClient', data3);
+    return;
+  }
```

Comment:
Let's extract the stuff inside this if statement into a function to reduce duplication.